### PR TITLE
ffa: Allow multiple SPs with same UUID

### PR DESCRIPTION
**Pull request by jellesels-arm**
Change the naming scheme of the SP binaries to make sure we can add multiple SPs with the same UUID to the system. 
This is need to support [PR](https://github.com/OP-TEE/optee_os/pull/5650)

### CODE REVIEW

**Comment by jbech-linaro** [comment]
Perhaps this has been discussed in the past, but why do we want to allow multiple SP's to have the same UUID in the first place? It's sounds wrong to me, but still I can see that this might make it easier from a (non-secure) user client point of view. I.e, a client sees "an interface" that belongs to a certain UUID, but under the hood on the secure side it has been split up into different SPs.

File: fvp/spmc_manifest.dts [hunk 1]
```diff
@@ -63,6 +64,25 @@
 			uuid = <0x33d532ed 0x0942e699 0x722dc09c 0xa798d9cd>;
 			load-address = <0x0 0x7e00000>;
 		};
+
+#else /*BUILD_SPMC_TESTS*/
```
Comment (etienne-lms) [inline fvp/spmc_manifest.dts:68]: update according to line 40.

File: fvp/spmc_manifest.dts [hunk 2]
```diff
@@ -37,6 +37,7 @@
 #ifdef ARM_BL2_SP_LIST_DTS
 	sp_packages {
 		compatible = "arm,sp_pkg";
+#if !SPMC_TESTS
```
Comment (etienne-lms) [inline fvp/spmc_manifest.dts:40]: `#ifndef CFG_SPMC_TESTS`?

File: fvp/bl2_sp_list.dtsi [hunk 3]
```diff
@@ -3,6 +3,7 @@
  * Copyright (c) 2022, Arm Limited. All rights reserved.
  */
 
+#if !SPMC_TESTS
```
Comment (etienne-lms) [inline fvp/bl2_sp_list.dtsi:6]: `ifndef CFG_SPMC_TESTS`?

File: fvp/bl2_sp_list.dtsi [hunk 4]
```diff
@@ -48,4 +48,9 @@ test_sp3 {
 	load-address = <0x7a20000>;
 };
 
+test_sp4 {
+	/* FF-A UUID */
```
Comment (etienne-lms) [inline fvp/bl2_sp_list.dtsi:52]: remove this inline comment? (for consistency within this DTSI file)

File: fvp/spmc_manifest.dts [hunk 5]
```diff
@@ -81,6 +81,11 @@
 			load-address = <0x0 0x7a20000>;
 		};
 
+		test_sp4 {
+			/* SP binary UUID */
```
Comment (etienne-lms) [inline fvp/spmc_manifest.dts:85]: Ditto

**Review by etienne-lms — COMMENTED** [review]
Commit "FF-A: Add test SPs to the fip package", in commit message:
s/fip/FIP/g
s/include/includes/
s/doesn't has/doesn't have/


Commit "FF-A: Add spm-test4 test sp", in commit message.
"... ~~it's~~ **its** binary ~~then them~~ **than** the one ..."

For both commits I suggest to use "fvp-psa-sp: " instead of "FFA: " in commit message header line.

File: fvp/spmc_manifest.dts (hunk 1 above)
Comment (jellesels-arm) [reply fvp/spmc_manifest.dts]: Ack

File: fvp/spmc_manifest.dts (hunk 2 above)
Comment (jellesels-arm) [reply fvp/spmc_manifest.dts]: Ack

File: fvp/bl2_sp_list.dtsi (hunk 3 above)
Comment (jellesels-arm) [reply fvp/bl2_sp_list.dtsi]: Ack

File: fvp/bl2_sp_list.dtsi (hunk 4 above)
Comment (jellesels-arm) [reply fvp/bl2_sp_list.dtsi:52]: Ack

File: fvp/spmc_manifest.dts (hunk 5 above)
Comment (jellesels-arm) [reply fvp/spmc_manifest.dts:85]: Ack

**Pull request closed without merging**